code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -22,22 +22,28 @@
 		gulp_size = max(round(reagents.total_volume / 5), 5)
 
 /obj/item/reagent_containers/food/drinks/attack(mob/living/M, mob/user, def_zone)
-
 	if(!reagents || !reagents.total_volume)
-		to_chat(user, "<span class='warning'>[src] is empty!</span>")
-		return 0
+		if(iscyborg(user))
+			synthetize_reagents() //how the fuck do I call this
```

Comment:
You need to make the object with the proc call the proc. Without the dot operator, the containing object (the drink glass here) will call it. You'll probably want to cast the user as a cyborg and make the cyborg call the proc with the dot operator.
>var/mob/living/carbon/silicon/robot/R = user
>R.synthesize_reagents()

Reply:
Got it. I did that on my branch already. I presume I have to do the same on the analyze one, right?

Reply:
I think I managed to fix this. I made it similar to the rechargstation code.